### PR TITLE
Fixes #8092

### LINKED ISSUE #8092: rdDetermineBonds segfaults when called from readonly directory
This was reported in #8091

The underlying code tries to make the files "nul" and "run.out".  On windows "nul" is the null file, but this is not true on linux.

Example, on linux cd to /usr and run this code

```
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdDetermineBonds
import os

import tempfile
from stat import S_IREAD, S_IRGRP, S_IROTH

m = Chem.AddHs(Chem.MolFromSmiles("CCC"))

AllChem.EmbedMolecule(m)
m = Chem.RWMol(m)
m.RemoveBond(0,1)
m.RemoveBond(1,2)

curdir = os.path.abspath(os.curdir)
with tempfile.TemporaryDirectory() as d:
    ## make temporary directory read only
    os.chdir(d)
    os.chmod(d, S_IREAD|S_IRGRP|S_IROTH)
    try:
        rdDetermineBonds.DetermineConnectivity(m, useHueckel=True)
    finally:
        os.chdir(curdir)
```

Results in 

```
segmentation fault  python 
```

### PR DESCRIPTION
**Pull request by greglandrum**
Creates the temp files in the system temporary directory instead of locally and gives them random names.

We might think about moving the RAII tempfile class somewhere else so that it's usable from other code

### CODE REVIEW

File: External/YAeHMOP/EHTTools.cpp [hunk 1]
```diff
@@ -25,6 +27,42 @@ const std::string _EHTChargeMatrix = "_EHTChargeMatrix";
...
+  return res;
+}
+// RAII tempfile class
+struct Tempfile {
+  std::filesystem::path fname;
+  FILE *fptr;
+
+  Tempfile() {
+    fname = std::filesystem::temp_directory_path() / randomstring();
+    fptr = std::fopen(fname.c_str(), "w+");
```
Comment (bp-kelley) [inline External/YAeHMOP/EHTTools.cpp:51]: We should probably add an exception

```
if (!fptr) {
 throw BadFileException("...");
}
```

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: External/YAeHMOP/EHTTools.cpp (hunk 1 above)
Comment (bp-kelley) [inline External/YAeHMOP/EHTTools.cpp:51]: Windows paths are wchar_t so can't be used like this in std::fopen

You might need to use _wfopen

```
fptr = _wfopen(fname..c_str(),  L"w+");
```

 or try and convert to char:

```
fptr = std::fopen(fname.string().c_str(), "w+");
```

**Review by bp-kelley — CHANGES_REQUESTED** [review]
Windows Updates

File: External/YAeHMOP/EHTTools.cpp (hunk 1 above)
Comment (bp-kelley) [reply External/YAeHMOP/EHTTools.cpp:51]: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170

Comment (greglandrum) [reply External/YAeHMOP/EHTTools.cpp]: done

**Review by bp-kelley — APPROVED** [review]

**Pull request merged**